x/ccv/staking/module.go

```diff
@@ -0,0 +1,53 @@
+package staking
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var (
+	_ module.AppModule           = AppModule{}
+	_ module.AppModuleBasic      = AppModuleBasic{}
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+// AppModule embeds the Cosmos SDK's x/staking AppModuleBasic.
+type AppModuleBasic struct {
+	staking.AppModuleBasic
+}
+
+// AppModule embeds the Cosmos SDK's x/staking AppModule where we only override
+// specific methods.
+type AppModule struct {
+	// embed the Cosmos SDK's x/staking AppModule
+	staking.AppModule
+}
+
+// NewAppModule creates a new AppModule object using the native x/staking module
+// AppModule constructor.
+func NewAppModule(cdc codec.Codec, keeper keeper.Keeper, ak types.AccountKeeper, bk types.BankKeeper) AppModule {
```

Comment:
if possible could you override the module keyname here?

Reply:
The module name? I _think_ so, but why would you want to anyway?

Reply:
A serious problem would arise if **_other_** modules depend on the module name, e.g. `bk.SendFrom*ToModule(..., staking.ModuleName)`.

Reply:
I'm wondering whether this name collision could be an issue on the consumer chain, since the slashing module replaces the staking module with the CCV module, while other modules, like governance, with still use the actual staking module (or more precisely, this modified version of it). 

@sainoe What do you think?